base/regex.jl

```diff
@@ -152,8 +152,10 @@ function match(re::Regex, str::UTF8String, idx::Integer, add_opts::UInt32=UInt32
     RegexMatch(mat, cap, ovec[1]+1, off, re)
 end
 
+_utf8(str) = utf8(str)
```

Comment:
@StefanKarpinski Even without the PCRE2 support for utf16 and utf32 (which I was also thinking of putting in a package), a slower approach would be to simply use convert(UTF8String, str) with the current PCRE (that was part of the reason I've been so concerned about the speed of conversions in Julia).